src/services/sync/strategies/TSP/TSPSchoolSyncer.js

```diff
@@ -352,21 +354,41 @@ class TSPSchoolSyncer extends mix(Syncer).with(ClassImporter) {
 	 */
 	async updateTeacher(user, tspTeacher) {
 		try {
+			const userUpdateObject = {};
+
+			// Check if anything has changed on the TSP side regarding user data
+			// and, if yes, set all the current data in the user update object.
 			const equal =
 				(user.namePrefix === tspTeacher.lehrerTitel || (!user.namePrefix && !tspTeacher.namePrefix)) &&
 				user.firstName === tspTeacher.lehrerVorname &&
 				user.lastName === tspTeacher.lehrerNachname;
-			if (equal) {
+			if (!equal) {
+				userUpdateObject.namePrefix = tspTeacher.lehrerTitel;
+				userUpdateObject.firstName = tspTeacher.lehrerVorname;
+				userUpdateObject.lastName = tspTeacher.lehrerNachname;
+			}
+
+			// If the feature flag is enabled, add the last synced at field
+			// to the user options (for the updated user document) to set
+			// the last sync date value to the current date.
+			if (lastSyncedAtEnabled) {
```

Comment:
better have that as a class prop instead of global

Reply:
Agree, changed that in [this commit](https://github.com/hpi-schul-cloud/schulcloud-server/pull/4221/commits/797b98d16278ac55848a410467d5fd1f8e04319c).